openwisp_monitoring/monitoring/tests/test_notifications.py

```diff
@@ -78,6 +78,67 @@ def test_general_check_threshold_deferred_not_crossed(self):
         self.assertTrue(m.is_healthy)
         self.assertEqual(Notification.objects.count(), 0)
 
+    def test_resources_metric_threshold_deferred_not_crossed(self):
+        # Verify that newly created metrics within threshold limit don't send any notifications
+        self._create_admin()
+        self.create_test_adata()
+        self.assertEqual(Notification.objects.count(), 0)
+
+    def test_cpu_metric_threshold_crossed(self):
+        admin = self._create_admin()
+        test_data_path = (
+            'openwisp_monitoring.device.tests.DeviceMonitoringTestCase._data'
+        )
+        data = self._data()
+        data['resources']['load'] = [0.99, 0.99, 0.99]
+        with patch(test_data_path, return_value=data):
+            self.create_test_adata()
+            m = Metric.objects.get(name='CPU usage')
+            self.assertEqual(Notification.objects.count(), 1)
+            n = Notification.objects.first()
+            self.assertEqual(n.recipient, admin)
+            self.assertEqual(n.actor, m)
+            self.assertEqual(n.action_object, m.threshold)
+            self.assertEqual(n.level, 'warning')
+
+    def test_memory_metric_threshold_crossed(self):
+        admin = self._create_admin()
+        test_data_path = (
+            'openwisp_monitoring.device.tests.DeviceMonitoringTestCase._data'
+        )
+        data = self._data()
+        data['resources']['memory']['available'] = 6000
+        data['resources']['memory']['free'] = 4000
+        with patch(test_data_path, return_value=data):
+            self.create_test_adata()
+            m = Metric.objects.get(name='Memory usage')
+            self.assertEqual(Notification.objects.count(), 1)
+            n = Notification.objects.first()
+            self.assertEqual(n.recipient, admin)
+            self.assertEqual(n.actor, m)
+            self.assertEqual(n.action_object, m.threshold)
+            self.assertEqual(n.level, 'warning')
+
+    def test_disk_metric_threshold_crossed(self):
```

Comment:
Not sure if we need to test each 'resource' metric, isn't it enough with just one?

Reply:
Though it is always better to test all cases but in this case it seems that including test for only one metric is sufficient to test the same for all three :smiley: